lib/fileset/default.nix

```diff
@@ -187,6 +188,24 @@ If a directory does not recursively contain any file, it is omitted from the sto
         filter = sourceFilter;
       };
 
+  /*
+  Create a file set from a filtered local source as produced by the `lib.sources` functions.
+  This does not import anything into the store.
+
+  Type:
+    fromSource :: SourceLike -> FileSet
+
+  Example:
+    fromSource (lib.sources.cleanSource ./.)
```

Comment:
The example in the PR description is also relevant, because it shows a bit of integration.

Reply:
Added some more practical examples now